wgpu-core/src/command/render.rs

```diff
@@ -382,13 +385,16 @@ struct VertexState {
 
 impl VertexState {
     fn update_limits(&mut self) {
-        self.vertex_limit = u32::MAX;
-        self.instance_limit = u32::MAX;
+        // Ensure that the limits are always smaller than u32::MAX so that
+        // interger overlows can be prevented via saturating additions.
+        let max = u32::MAX - 1;
+        self.vertex_limit = max;
+        self.instance_limit = max;
```

Comment:
isn't it valid to have a vertex limit of u32::MAX? Wouldn't that mean that the last vertex/instance isn't accounted for?

Reply:
I was also confused from the OP as to why `saturating_add` was selected, as opposed to `checked_add`. @nical, would that not be more appropriate?

Reply:
With this patch, having u32::MAX vertices would produce an error. It's a bit theoretical because you'll most likely hit the max buffer size before you manage to stuff u32::MAX vertices into the buffer.

Reply:
`checked_add` forces you to spell out an separate check and return path for this whereas `saturating_add` was a simple way to just treat the  "range too big" case in one go without having to deal with overflow separately. It's also hot code so keeping the validation short has some value.